src/sql/workbench/services/query/common/query.ts

```diff
@@ -75,3 +87,10 @@ export interface ICellValue {
 	isNull?: boolean;
 	invariantCultureDisplayValue?: string;
 }
+
+export interface IQueryResultsWriter {
+	enable(): void;
+	disable(): void;
```

Comment:
Generally easier just to have a single enabled property instead of two separate functions

Reply:
The reason behind the second property of `disable` is to unsubscribe a writer from the query runner when the write mode changes by disposing of the listeners added to the disposable store.